### DISCUSSION TITLE
Problem with wifi menu pywal theme

### DISCUSSION THREAD

**Discussion by emretuna** [davatorium/rofi#1676] [Q&A]
Hi,
Rofi Version: 1.7.1
I use wifi-menu from this repo:https://github.com/ericmurphyxyz/rofi-wifi-menu/blob/master/rofi-wifi-menu.sh

when I run it shows connections no problem.
I themed it and when I run with this command it shows themed window but empty list.What am I doing wrong?

theme:https://github.com/emretuna/.dotfiles/blob/main/wal/.config/wal/templates/wifimenu.rasi

`bindsym $mod+Shift+w exec rofi -modi 'Wifimenu:~/.config/rofi/scripts/wifimenu.sh' -show Wifimenu -theme '~/.cache/wal/wifimenu.rasi' -icon-theme "RosePine"`

**Comment by DaveDavenport** [comment]
I don't think we can answer this. You need to provide more information.
* the actual theme.
* rofi version

maybe this is more suitable for the pywal discussion forum, as I have no clue what that is or does.

**Reply by emretuna** [reply]
Actually, I already did and wanted to try my luck here too. Maybe someone has a clue or smth.

**Reply by DaveDavenport** [reply]
Again, you need to provide information if you want to get help.
How can I tell you what is wrong with the theme, if you do not provide the theme.

**Reply by emretuna** [reply]
Thanks, added it.

**Reply by DaveDavenport** [reply]
that is not a valid theme syntax.

please post the **rofi** theme.

**Reply by emretuna** [reply]
https://github.com/emretuna/.dotfiles all in this repo.

**Reply by DaveDavenport** [reply]
Sorry, if you do not provide the requested information, I will not help.

**Reply by emretuna** [reply]
Thanks for your efford, maybe someone who uses pywal will better understand the issue.
Do you have an idea why `' is neither SSID nor BSSID. rofi from inside rofi.` error occurs?

**Reply by DaveDavenport** [reply]
It is not an understanding issue, its that you are **not** providing the requested information.
I need to see the final theme (for the path, see your original post) to be able to see what is wrong.

**Reply by DaveDavenport** [reply]
>rofi from inside rofi.

you cannot launch rofi, from within rofi. the script you use is broken.

**Comment by emretuna** [comment]
> rofi -modi 'Wifimenu:~/.config/rofi/scripts/wifimenu.sh' -show Wifimenu -theme '~/.cache/wal/wifimenu.rasi'

Script working fine when I launch it from terminal like `./wifimenu.sh`
But I have to use this command `rofi -modi 'Wifimenu:~/.config/rofi/scripts/wifimenu.sh' -show Wifimenu -theme '~/.cache/wal/wifimenu.rasi'`
or define it inside wifimenu.sh to use it with template located in `~/.cache/wal/wifimenu.rasi`

**Reply by DaveDavenport** [reply]
If it works fine when launching from terminal, you cannot launch it as script mode!!!
In the first case, the script calls rofi (this is rofi dmenu compatibility mode), in the second rofi calls the script. This is not compatible with each other.


The file you want to use it with is not a template, that is the actual theme I needed to see.
But before debugging a theme, you need to use rofi correctly.

**Reply by emretuna** [reply]
That was all I need, I fixed it thanks a lot!
